src/drivers/sdl/lv_sdl_window.c

```diff
@@ -208,20 +209,46 @@ static void flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * px_m
 #if LV_USE_DRAW_SDL == 0
     lv_sdl_window_t * dsc = lv_display_get_driver_data(disp);
     if(LV_SDL_RENDER_MODE == LV_DISPLAY_RENDER_MODE_PARTIAL) {
-        int32_t y;
-        uint8_t * fb_tmp = dsc->fb_act;
+
+        uint8_t rotated_buf[32 * 1024];
+        lv_area_t rotated_area;
+        lv_display_rotation_t rotation = lv_display_get_rotation(disp);
+        if(rotation == LV_DISPLAY_ROTATION_90) {
+            int32_t w = lv_area_get_width(area);
+            int32_t h = lv_area_get_height(area);
+            lv_color_format_t cf = lv_display_get_color_format(disp);
+            lv_draw_sw_rotate(px_map, rotated_buf,
+                              w, h, lv_draw_buf_width_to_stride(w, cf),
+                              lv_draw_buf_width_to_stride(h, cf),
+                              LV_DISPLAY_ROTATION_90, cf);
+
+            rotated_area.x1 = area->y1;
+            rotated_area.y2 = disp->ver_res - area->x1 - 1;
+            rotated_area.x2 = rotated_area.x1 + h - 1;
+            rotated_area.y1 = rotated_area.y2 - w + 1;
```

Comment:
Maybe rotation an area could move into a convenience function similar to `lv_draw_sw_rotate` as it'll be needed in any display driver that wants to support software rotation?

Reply:
I agree that it's not too convenient now but it's flexible this way. If you need to call `lv_draw_sw_rotate` yourself you can call a custom function instead too. E.g. an ASM optimized version, or a GPU accelerated one.